mdsuite/transformations/thermal_flux.py

```diff
@@ -24,193 +24,71 @@
-
-        Parameters
-        ----------
-        data : tf.Tensor
-                Data on which to apply the operation.
-        Returns
-        -------
-        system_current : np.array
-                System current as a numpy array.
-        """
-
-        system_current = np.zeros((self.batch_size, 3))
-        for species in self.experiment.species:
-            stress_path = str.encode(join_path(species, "Stress"))
-            velocity_path = str.encode(join_path(species, "Velocities"))
-            ke_path = str.encode(join_path(species, "KE"))
-            pe_path = str.encode(join_path(species, "PE"))
+        super(ThermalFlux, self).__init__(
+            input_properties=[
+                mdsuite_properties.stress,
+                mdsuite_properties.velocities,
+                mdsuite_properties.kinetic_energy,
+                mdsuite_properties.potential_energy,
+            ],
+            output_property=mdsuite_properties.thermal_flux,
+            batchable_axes=[1, 2],
+            scale_function={"linear": {"scale_factor": 5}},
+        )
+
+    def transform_batch(
+        self, batch: typing.Dict[str, typing.Dict[str, tf.Tensor]]
+    ) -> tf.Tensor:
+        fluxes = []
+        for _, properties in batch.items():
+            stress = properties[mdsuite_properties.stress.name]
+            vel = properties[mdsuite_properties.velocities.name]
+            ke = properties[mdsuite_properties.kinetic_energy.name]
+            pe = properties[mdsuite_properties.potential_energy.name]
             phi_x = (
-                np.multiply(data[stress_path][:, :, 0], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 3], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 4], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 0], vel[:, :, 0])
+                + np.multiply(stress[:, :, 3], vel[:, :, 1])
+                + np.multiply(stress[:, :, 4], vel[:, :, 2])
             )
             phi_y = (
-                np.multiply(data[stress_path][:, :, 3], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 1], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 5], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 3], vel[:, :, 0])
+                + np.multiply(stress[:, :, 1], vel[:, :, 1])
+                + np.multiply(stress[:, :, 5], vel[:, :, 2])
             )
             phi_z = (
-                np.multiply(data[stress_path][:, :, 4], data[velocity_path][:, :, 0])
-                + np.multiply(data[stress_path][:, :, 5], data[velocity_path][:, :, 1])
-                + np.multiply(data[stress_path][:, :, 2], data[velocity_path][:, :, 2])
+                np.multiply(stress[:, :, 4], vel[:, :, 0])
+                + np.multiply(stress[:, :, 5], vel[:, :, 1])
+                + np.multiply(stress[:, :, 2], vel[:, :, 2])
```

Comment:
I think `stress[:, :, 0] * vel[:, :, 0]` is easier to read than `np.multiply(stress[:, :, 0], vel[:, :, 0])` also feels like this could be mor vectorized but I would have to look up the formulas behind it. It is not really part of the PR but can be quickly changed.

Reply:
I think if the data is all tf.Tensor then using np might actually slow it down unless tf automatically does the conversion.

Reply:
I changed it to standard tf multiplication